File: DataStructure/LinkedList/LinkedList.java
package DataStructure.LinkedList;
/*
Note: 

Keep the usage of in-built function to the minimum
Add/Optimize  features
Please follow best coding practices
Add comments if necessary
And make the Code Bug free

*/

class Node {
    int data;
    Node next;
    Node( int n ) {
        data = n;
        next = null;
    }
}

public class LinkedList {
    
    int size;
    Node root;

    LinkedList(){
        root = null;
        size = 0;
    }

    void add( int num ) {
        Node newNode = new Node( num );
        newNode.next = root;
        root = newNode;
        System.out.println( "Added: " + root.data ) ;
        size++;
    }

    // Add appropriate message when the list is empty
    void del() {
        System.out.println( "Deleted: " + root.data ) ;
        root = root.next;
        size--;
    }

    Node elementAt( int index ) {
    //     LinkedList: 10 <- 20 <- 30 <- 40
    //     Index:      0     1     2     3
    
        int len = size-1;
        Node temp = root;

        while( index != len ) {
            temp = temp.next;
            len--;   
        }

        return temp;

    }

    void print() {
        Node temp = root;
        System.out.print( "LinkedList Values: " );
        while( temp != null ) {
            System.out.print( temp.data + " " );
            temp = temp.next;
        }
        System.out.println() ;
    }

    // Test and Complete the following methods

    /*
    void printReverse(){
    
    }

    void reverse() {
        // Reverse the content of the list

    }

    void sort(){
        // sort elements of list ( ascending )

    }

    void clear() {
        // Delete all the elements from the list
    }

    bool contains( int value ) {
        // To check if element exist
    }

    */
    public static void main(String[] args) {

        LinkedList ll = new LinkedList();
        ll.add( 1 );
        ll.print();
    
        ll.add( 2 );
        ll.print();
        
        ll.del();
        ll.print();
    
        ll.add(3);
        ll.print();
        
        int ind = 1 ;
        Node node = ll.elementAt( ind );
        System.out.println( "Element at index " + ind + ": " + node.data );        

    }

}

File: DataStructure/LinkedList/LinkedList.py
# Note: 

# Keep the usage of in-built function to the minimum
# Add/Optimize  features
# Please follow best coding practices
# Add comments if necessary
# And make the Code Bug free


class Node:
    def __init__( self, num ):
        self.data = num
        self.next = None

class LinkedList:
    def __init__( self ):
        self.root = None
        self.size = 0

    def add( self, num ):
        newNode = Node( num )
        newNode.next = self.root
        self.root = newNode
        print( "Added:" , self.root.data ) 
        self.size += 1

    # Add appropriate message when the list is empty
    def delete( self ):
        print( "Deleted:" , self.root.data ) 
        self.root = self.root.next
        self.size -= 1

    def elementAt( self, index ):
    #     LinkedList: 10 <- 20 <- 30 <- 40
    #     Index:      0     1     2     3
    
        length = self.size-1
        temp = self.root

        while( index != length ):
            temp = temp.next
            length -= 1   

        return temp

    def print( self ) :
        temp = self.root
        print( "LinkedList Values: ", end="" )
        while( temp != None ):
            print( temp.data , end=" ")
            temp = temp.next
        print()

    # Test and Complete the following methods

    
    def printReverse( self ):
        pass

    def reverse( self ): 
        # Reverse the content of the list
        pass
    def sort( self ):
        # sort elements of list ( ascending )
        pass

    def clear( self ) :
        # Delete all the elements from the list
        pass

    def contains( self, value ) :
        # To check if element exist
        pass

ll = LinkedList()
ll.add( 1 )
ll.print()

ll.add( 2 )
ll.print()

ll.delete()
ll.print()

ll.add(3)
ll.print()

ind = 1 
node = ll.elementAt( ind )
print( "Element at index" , ind , ":" , node.data )       
